include/rosdiscover-clang/BackwardSymbolizer/FunctionSymbolizer.h

```diff
@@ -687,8 +690,135 @@ class FunctionSymbolizer {
+            // inner statement. Hence those edges need to be ignored to avoid circles in the control dependcy graph. 
+            if ((postdominatorAnalysis.dominates(depBlock->getClangBlock(), sBlock) 
+              && !dominatorAnalysis.dominates(depBlock->getClangBlock(), sBlock)) 
+              || depBlock->getClangBlock()->getBlockID() == sBlock->getBlockID()) {
+              if (i == 0) { //true branch, as defined by clang's order of successors                
+                  llvm::outs() << "true branch dominates stmt\n";
+                  trueBranchDominates = true;
+              } else if (i == 1) { //false branch
+                  llvm::outs() << "false branch dominates stmt\n";
+                  falseBranchDominates = true;
+              } 
+            } 
+            if (i > 1) {
+              //TODO: Handle switch-case here.
+              llvm::outs() << "Too many branches. Swich not yet supported\n";
+              abort();
+            }
+            i++;
+          }
+          if (!trueBranchDominates && !falseBranchDominates) {
+            continue; // No edge needed
+          }
+          if (trueBranchDominates && falseBranchDominates){
+            llvm::outs() << "ERROR: falseBranchDominates: " << falseBranchDominates << " trueBranchDominates: " << trueBranchDominates << "\n";
+            llvm::outs() << "depBlock->getClangBlock(): ";
+            depBlock->getClangBlock()->dump();
+            llvm::outs() << "\npredecessor->getClangBlock(): ";
+            predecessor->getClangBlock()->dump();
+            abort();
+          }
+          
+          // Creating edge
+          CFGEdge::EdgeType type;
+          if (i == 1) {
+            type = CFGEdge::EdgeType::Normal;
+          } else if (i == 2) {
+            type = falseBranchDominates ? CFGEdge::EdgeType::False : CFGEdge::EdgeType::True;
+          } else {
+            type = CFGEdge::EdgeType::Unknown;
+            llvm::outs() << "ERROR: Unknown edge type\n";
+          }
+
+          if (predecessor->createEdge(depBlock, type)) {
+            llvm::outs() << "created edge between " << predecessor->getConditionStr(astContext) << " and " << depBlock->getConditionStr(astContext) << " of type " << CFGEdge::getEdgeTypeName(type) << "\n";
+          }
+        }
+      }
+      return {newControlDependencyNode}; // Return newly created control dependency node
+    } else { 
+      return predecessors; // For non-control dependency nodes forward the predecessors to the next dependency node
+    }
+  }
+
+  CFGBlock* buildGraph(const clang::CFGBlock* blockOfInterest, const llvm::SmallVector<clang::CFGBlock *, 4> &deps, clang::CFGDominatorTreeImpl<true> &postdominatorAnalysis, clang::CFGDominatorTreeImpl<false> &dominatorAnalysis) {
+    std::vector<const clang::CFGBlock*> analyzed;
+    std::unordered_map<long, CFGBlock*> blockMap; //maps BlockID to CFGBlockObject
```

Comment:
suggest renaming to `idToBlock` to avoid need for comment

Reply:
fixed